src/base/QXmppMixItem.cpp

```diff
@@ -0,0 +1,166 @@
+ *  https://github.com/qxmpp-project/qxmpp
+ *
+ * This file is a part of QXmpp library.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ */
+
+#include "QXmppMixItem.h"
+#include "QXmppDataForm.h"
+#include "QXmppConstants_p.h"
+#include "QXmppUtils.h"
+#include <QDomElement>
+#include <QBuffer>
+
+QString QXmppMixInfoItem::name() const
+{
+    return m_name;
+}
+
+void QXmppMixInfoItem::setName(const QString& name)
+{
+    m_name = name;
+}
+
+QString QXmppMixInfoItem::description() const
+{
+    return m_description;
+}
+
+void QXmppMixInfoItem::setDescription(const QString& description)
+{
+    m_description = description;
+}
+
+QStringList QXmppMixInfoItem::contactJids() const
+{
+    return m_contactJids;
+}
+
+void QXmppMixInfoItem::setContactJids(const QStringList& contactJids)
+{
+    m_contactJids = contactJids;
+}
+
+bool QXmppMixInfoItem::isMixChannelInfo(const QDomElement& element)
+{
+    return element.tagName() == "item" && element.hasAttribute("id")
+           && element.namespaceURI() == ns_mix;
+}
+
+void QXmppMixInfoItem::parse(const QXmppElement& element)
+{
+    QXmppDataForm form;
+    form.parse(element.sourceDomElement());
+
+    for (auto& field : form.fields()) {
+        if (field.key() == "Name")
+            m_name = field.value().toString();
+        else if (field.key() == "Description")
+            m_description = field.value().toString();
+        else if (field.key() == "Contact")
+            m_contactJids = field.value().toStringList();
+    }
+}
+
+QXmppElement QXmppMixInfoItem::toElement() const
+{
+    QXmppDataForm form;
+    form.setType(QXmppDataForm::Result);
+    QList<QXmppDataForm::Field> fields;
+
+    QXmppDataForm::Field formType;
+    formType.setType(QXmppDataForm::Field::HiddenField);
+    formType.setKey("FORM_TYPE");
+    formType.setValue(ns_mix);
+    fields << formType;
+
+    QXmppDataForm::Field nameField;
+    nameField.setKey("Name");
+    nameField.setValue(m_name);
+    fields << nameField;
+
+    QXmppDataForm::Field descriptionField;
+    descriptionField.setKey("Description");
+    descriptionField.setValue(m_description);
+    fields << descriptionField;
+
+    QXmppDataForm::Field contactsField;
+    contactsField.setKey("Contact");
+    contactsField.setValue(m_contactJids);
+    contactsField.setType(QXmppDataForm::Field::JidMultiField);
+    fields << contactsField;
+
+    form.setFields(fields);
+
+    // FIXME: this is too complicated; maybe don't use QXmppElement in QXmppPubSubItem?
```

Comment:
Another solution would be to use QXmppElement for everything; parsing and serialization. But this and using a proper JID class would probably both be tasks for "QXmpp 2.0", although it were theoretically possible to keep compatibility by creating all methods with both parameter types.

Reply:
I think I'd rather see QXmppElement die that expanding its usage